app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/recentactivity/RecentActivityFragment.kt

```diff
@@ -42,37 +42,53 @@ class RecentActivityFragment : Fragment(), RecentActivityContract.View {
 
         recentActivityRecyclerAdapter = RecentActivityRecyclerAdapter(R.layout.item_rv_recent_activity)
 
+        initializeRecyclerView()
+        setData()
+        showProgressBar()
+        showMessage()
+    }
+
+    private fun initializeRecyclerView() {
         recyclerEditedArticlesList.apply {
             layoutManager = LinearLayoutManager(context)
             setHasFixedSize(true)
             adapter = recentActivityRecyclerAdapter
         }
-
-        url?.let { recentActivityPresenter.requestRecentActivity(it) }
     }
 
-    override fun setData(data: RecentActivityResponse) {
-        Timber.d(data.toString())
-        if (data.course.revisions.isNotEmpty()) {
-            recyclerEditedArticlesList?.visibility = View.VISIBLE
-            recentActivityRecyclerAdapter.setData(data.course.revisions)
-            recentActivityRecyclerAdapter.notifyDataSetChanged()
-            textViewNoActivity?.visibility = View.GONE
-        } else {
-            recyclerEditedArticlesList?.visibility = View.GONE
-            textViewNoActivity?.visibility = View.VISIBLE
-        }
+    /**
+     *   This set the data on the view from the viewmodel
+     */
+    fun setData() {
+        recentActivityViewModel.data.observe(this, Observer {
+            Timber.d("recent $it.size")
+            if (it.isNotEmpty()) {
+                recyclerEditedArticlesList?.visibility = View.VISIBLE
+                recentActivityRecyclerAdapter.setData(it)
+                textViewNoActivity?.visibility = View.GONE
+            } else {
+                recyclerEditedArticlesList?.visibility = View.GONE
+                textViewNoActivity?.visibility = View.VISIBLE
+            }
+        })
     }
 
-    override fun showProgressBar(show: Boolean) {
-        progressBar?.visibility = if (show) {
-            View.VISIBLE
-        } else {
-            View.GONE
-        }
+    /**
+     *   This shows the progressbar
+     */
+    fun showProgressBar() {
```

Comment:
showProgressBar -> initialzeProgressBar as we are setting observing the change instead of showing/hiding progress-bar .

Reply:
Not too clear

Reply:
Done

Reply:
functionName refactor to 'initialzeProgressBar' instead of 'showProgressBar' .